packages/hardhat/test/e2e/periphery/controller-helper.ts

```diff
@@ -1356,4 +1460,635 @@ it("Close vault LP and open new one-siced LP with just oSQTH ", async () => {
+      console.log('wPowerPerpAmountInLPBefore', wPowerPerpAmountInLPBefore.toString())
+      console.log('wethAmountInLPBefore', wethAmountInLPBefore.toString())
+      console.log('wPowerPerpAmountInLPAfter', wPowerPerpAmountInLPAfter.toString())
+      console.log('wethAmountInLPAfter', wethAmountInLPAfter.toString())
+      console.log('vaultSqueethDiff',vaultSqueethDiff.toString() )
+      console.log('vaultEthDiff',vaultEthDiff.toString() )
+      console.log('lpSqueethDiff', lpSqueethDiff.toString())
+      console.log('lpEthDiff', lpEthDiff.toString())
+      console.log('depositorSqueethDiff', depositorSqueethDiff.toString())
+      console.log('vaultBefore.collateralAmount', vaultBefore.collateralAmount.toString())
+      console.log('vaultBefore.shortAmount', vaultBefore.shortAmount.toString())
+      console.log('vaultAfter.collateralAmount', vaultAfter.collateralAmount.toString())
+      console.log('vaultAfter.shortAmount', vaultAfter.shortAmount.toString())
+      console.log('flashloanAmount',flashLoanAmount.toString())
+      console.log('wPowerPerpAmountInLPBefore', wPowerPerpAmountInLPBefore.toString())
+      console.log('depositorEthDiff', depositorEthDiff.toString())
+      console.log('depositorSqueethDiff', depositorSqueethDiff.toString())
+      console.log('wPowerPerpToSwap', wPowerPerpToSwap.toString())
+      
+      // Swap amount removed from LP
+      expect(vaultSqueethDiff.sub(wPowerPerpToSwap).eq(0)).to.be.true
+      // Eth deposited = target amount
+      expect(vaultEthDiff.sub(ethAmountToDeposit.sub(ethers.utils.parseUnits('0.1'))).eq(0)).to.be.true
+      // Nft id unchanged
+      expect(vaultAfter.NftCollateralId==vaultBefore.NftCollateralId).to.be.true
+        })
+      })
+    
+      describe("Rebalance LP in vault: reduce oSQTH > debt, burn, withdraw >0 eth", async () => {
+        
+        before("Flashmint new LP with 50 oSQTH", async () => {
+          const normFactor = await controller.getExpectedNormalizationFactor()
+          const mintWSqueethAmount = ethers.utils.parseUnits('200')
+          const mintRSqueethAmount = mintWSqueethAmount.mul(normFactor).div(one)
+          const ethPrice = await oracle.getTwap(ethUsdcPool.address, weth.address, usdc.address, 420, true)
+          const scaledEthPrice = ethPrice.div(10000)
+          const debtInEth = mintRSqueethAmount.mul(scaledEthPrice).div(one)
+          const collateralToDeposit = debtInEth.mul(3).div(2).add(ethers.utils.parseUnits('0.01'))
+          const squeethPrice = await oracle.getTwap(wSqueethPool.address, wSqueeth.address, weth.address, 420, true)
+          const collateralToLp = mintWSqueethAmount.mul(squeethPrice).div(one)
+          const flashloanWMintDepositNftParams = {
+            vaultId: 0,
+            wPowerPerpAmount: mintWSqueethAmount,
+            collateralToDeposit: collateralToDeposit,
+            collateralToFlashloan: collateralToDeposit,
```

Comment:
no flashloan needed here, as you are depositing the correct amount of collateral for it to be collateralized correctly